Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectBuilder/MSBuildProjectBuilder.cs

```diff
@@ -365,14 +370,22 @@ private static async Task<int> BuildProjectAsync(string projectPath, BuildEngine
             switch (buildEngine)
             {
                 case BuildEngine.DotNet:
-                    msBuildPath = "dotnet";
+                    if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+                    {
+                        msBuildPath = "/usr/local/share/dotnet/dotnet";
+                    }
+                    else
+                    {
+                        msBuildPath = "dotnet";
+                    }
                     arguments = $"msbuild {arguments}";
                     break;
 
                 case BuildEngine.VisualStudio:
                     if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
```

Comment:
>if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX)) [](start = 20, length = 52)

NIT: In many environments 'msbuild' will already be an available command (presumably why this was working originally).  This also provides the ability for selecting an msbuild that is not the latest installed version.  You can check the environment variable PATH to see if it already has a path to the msbuild command and then not have to use the full path.

NIT: If going for the simplest path, instead of Current/Versions you can use the shortcuts directly in Mono.framework/Commands/, which by default will point to latest, but also provide the ability for the user to configure a different version as the 'default'.

As an example of both, here's some code using both of these recommendations: [https://microsoft.visualstudio.com/Apps/_git/MRMW.Monterrey?path=%2FMonterrey.Unity%2FAssets%2FBuild%2FEditor%2FRegenerateResourceFiles.cs&version=GBmaster&line=87&lineStyle=plain&lineEnd=87&lineStartColumn=21&lineEndColumn=50](https://microsoft.visualstudio.com/Apps/_git/MRMW.Monterrey?path=%2FMonterrey.Unity%2FAssets%2FBuild%2FEditor%2FRegenerateResourceFiles.cs&version=GBmaster&line=87&lineStyle=plain&lineEnd=87&lineStartColumn=21&lineEndColumn=50)

Reply:
So this wasn't working originally on the mac :) But, perhaps we could chat about how it should be working and see why it wasn't?